app/models/line_item.rb

```diff
@@ -1,5 +1,6 @@
 class LineItem < ApplicationRecord
-  belongs_to :product
+  belongs_to :order, optional: true
+  belongs_to :product, optional: true
```

Comment:
I think you intended the `optional: true` value to be set on  the `cart` association. That's the way the book has it written, anyway.

Reply:
Shit, thanks, good catch :)